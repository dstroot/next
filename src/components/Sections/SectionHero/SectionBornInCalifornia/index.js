import React from 'react';
// component
import SectionHero from '../../SectionHero';
import Clock from '../SectionBornInCalifornia/media/Clock';

// image

import bg from './media/Office.svg';

// styles
const background = {
  /* Image  */
  backgroundImage: `url(${bg})`,
};

const SectionBornInCalifornia = () => {
  return (
    <>
      <Clock />
      <SectionHero
        backgroundImage={background}
        heading="Born in California"
        content="Insurance products to complement your life goals and passions. We help you go after whatever's next for you."
      />
    </>
  );
};

export default SectionBornInCalifornia;
