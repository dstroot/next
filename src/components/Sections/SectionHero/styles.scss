// styles
.background {
  /* Set height relative to width */
  height: 100vh;
  min-height: 500px;
  width: 100%;

  /* Position the image to scale nicely on all screens */
  background-position: bottom;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;

  display: table;

  /* bottom border */
  border-bottom: 3px solid #dde1e7;
}

// place hero image on the background
.hero-image {
  width: 20vw;
  min-width: 175px;
  position: absolute;
  bottom: 0;
  left: 10%;
}

// place the text on the background image
.divText {
  display: table-cell;
  vertical-align: middle;
}
