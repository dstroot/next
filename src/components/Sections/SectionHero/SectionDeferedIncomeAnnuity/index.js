import React, { useEffect } from 'react';
import AnchorLink from 'react-anchor-link-smooth-scroll';

//component
import SectionHero from '../../SectionHero';
import Mountainbike from '../SectionDeferedIncomeAnnuity/media/Mountainbike';

//image
import mainbgforestlake3 from '../../SectionHero/SectionDeferedIncomeAnnuity/media/mainbgforestlake3.svg';
import mountainbike from '../../SectionHero/SectionDeferedIncomeAnnuity/media/mountainbike.svg';
import arrow from './media/arrowdown.svg';

const background = {
  /* Image  */
  backgroundImage: `url(${mainbgforestlake3})`,
};

const SectiionDeferedIncomeAnnuity = () => {
  useEffect(() => {
    window.scrollTo(0, 0);
  });
  return (
    <>
      <Mountainbike />
      <SectionHero
        backgroundImage={background}
        heading={
          <h1 className="nxt_heading-xlarge">Next Deferred Income Annuity</h1>
        }
        content="A predictable and reliable paycheck for life."
        heroImage={mountainbike}
        linkElement={
          <AnchorLink href="#stayintouch" className="text-decoration-none mt-4">
            <p className="mt-4 mb-2 text-uppercase text-primary">Learn More</p>
            <img
              className="d-block-inline"
              width={45}
              src={arrow}
              alt="Arrow"
            />
          </AnchorLink>
        }
        heroImageAlt="Mountain Bike"
      />
    </>
  );
};

export default SectiionDeferedIncomeAnnuity;
