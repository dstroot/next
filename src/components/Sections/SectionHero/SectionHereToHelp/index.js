import React from 'react';
// component
import SectionHero from '../../SectionHero';

// image

import bg from './media/contactladys.svg';
import twoLights from './media/twolights.svg';

// styles
const background = {
  /* Image  */
  backgroundImage: `url(${bg})`,
};

const lightStyle = {
  width: '42vh',
  minWidth: '50px',
  position: 'absolute',
  marginLeft: 'auto',
  marginRight: 'auto',
  marginTop: '12vh',
  left: '0',
  right: '0',
};

const SectionHereToHelp = () => {
  return (
    <>
      <img src={twoLights} style={lightStyle} alt="Two Lights" />
      <SectionHero
        backgroundImage={background}
        heading={
          <h1 className="nxt_heading-xlarge">
            Here to help whenever you need us!
          </h1>
        }
        content="Insurance products to complement your life goals and passions. We help you go after whatever's next for you."
      />
    </>
  );
};

export default SectionHereToHelp;
