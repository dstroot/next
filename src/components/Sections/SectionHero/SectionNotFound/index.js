import React from 'react';
// component
import SectionHero from '../../SectionHero';
import Button from '../../../Buttons/Button';

// image
// import notfound from './media/notfound.svg';
import bg from './media/notfound.svg';

// styles
const background = {
  /* Image  */
  backgroundImage: `url(${bg})`,
};

const SectionNotFound = () => {
  return (
    <>
      <SectionHero
        backgroundImage={background}
        heading={
          <h1 className="nxt_heading-xlarge">
            Sorry, we couldn’t find that page!
          </h1>
        }
        content="It’s ok to get lost sometimes. Let’s get you back on track."
        linkElement={<Button path="/" buttonText="Go to Homepage" />}
      />
    </>
  );
};

export default SectionNotFound;
