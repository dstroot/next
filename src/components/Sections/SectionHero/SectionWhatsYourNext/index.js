import React from 'react';
// component
import SectionHero from '../../SectionHero';

// image

import arrow from './media/arrowdown.svg';
import moped from './media/moped.svg';
import bg from './media/mainbg1animationB.svg';

// styles
const background = {
  /* Image  */
  backgroundImage: `url(${bg})`,
};

const SectionWhatsYourNext = () => {
  return (
    <SectionHero
      backgroundImage={background}
      heading="Whatâ€™s your next adventure?"
      content="Insurance products to complement your life goals and passions. We help you go after whatever's next for you."
      linkElement={
        <a href="#bestlife" className="text-decoration-none mt-4">
          <p className="mt-4 mb-2 text-uppercase text-primary">Learn More</p>
          <img className="d-block-inline" width={45} src={arrow} alt="Arrow" />
        </a>
      }
      heroImage={moped}
      heroImageAlt="moped"
    />
  );
};

export default SectionWhatsYourNext;
