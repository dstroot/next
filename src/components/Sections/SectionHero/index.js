import React from 'react';

// styles
import './styles.scss';

const SectionHero = props => {
  return (
    <div className="background" style={props.backgroundImage}>
      <div className="divText">
        <div className="container">
          <div className="row">
            <div className="col text-center mt-6">
              <h1 className="nxt_heading-large mt-5">{props.heading}</h1>
            </div>
          </div>
          <div className="row">
            <div className="col-md-8 offset-md-2 text-center">
              <p className="mt-4">{props.content}</p>
              {props.linkElement}
            </div>
          </div>
        </div>
      </div>
      {/* <img
        className="hero-image"
        src={props.heroImage}
        alt={props.heroImageAlt}
      /> */}
    </div>
  );
};

export default SectionHero;
