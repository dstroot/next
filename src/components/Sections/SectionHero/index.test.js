import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import SectionHero from '../SectionHero';

describe('SectionHero', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<SectionHero />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<SectionHero />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
