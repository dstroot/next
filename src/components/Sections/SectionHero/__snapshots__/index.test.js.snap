// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHero it should render 1`] = `
<div
  className="background"
>
  <div
    className="divText"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col text-center"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-8 offset-md-2 text-center"
        >
          <p
            className="mt-4 circular"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
